src/snapred/ui/workflow/DiffCalWorkflow.py

```diff
@@ -356,7 +361,17 @@ def _renewFitPeaks(self, peakFunction):
             detectorPeaks=self.ingredients.groupedPeakLists,
             peakFunction=peakFunction,
         )
-        return self.request(path="calibration/fitpeaks", payload=payload.json())
+        response = self.request(path="calibration/fitpeaks", payload=payload.json())
+        self._calculateResidual()
```

Comment:
I think this call should occur in L294 and L301 instead

Reply:
Done